app/controllers/projects/blob_controller.rb

```diff
@@ -146,10 +146,19 @@ def editor_variables
 
     @file_path =
       if action_name.to_s == 'create'
+        if params[:file_upload].present?
+          params[:file_name] = params[:file_upload].original_filename
+        end
         File.join(@path, File.basename(params[:file_name]))
       else
         @path
       end
+     
+    if params[:file_upload].present?
+      params[:content] = params[:file_upload].read
+      params[:encoding] = 'text'
```

Comment:
I guess a single file is okay for a first iteration. Maybe if the feature
is promoted again one may add more functionality?

@stanhu Can this be merged now?

I know, I sound a bit impatient... Sry for that.
--73
On Sep 8, 2015 12:31 PM, "Yakun Li" <notifications@github.com> wrote:

> In app/controllers/projects/blob_controller.rb
> <https://github.com/gitlabhq/gitlabhq/pull/9138#discussion_r38910985>:
>
> >          File.join(@path, File.basename(params[:file_name]))
> >        else
> >          @path
> >        end
> > +
> > +    if params[:file_upload].present?
> > +      params[:content] = params[:file_upload].read
> > +      params[:encoding] = 'text'
>
> @stanhu <https://github.com/stanhu> Uploading files through Dropzone is
> working properly now. I have some questions still,
> 1. Should we allow multiple files upload?
> 2. If we allow multiple files upload, then should we redirect user to the
> tree view rather than blob view after the files are uploaded?
>
> Thanks.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gitlabhq/gitlabhq/pull/9138/files#r38910985>.
>